### PR TITLE
Switches to checking for None with "is" instead of Count.

### LINKED ISSUE #350: CrossValidation does not work with newer numpy.
--------
Issue Description
--------
##### What did you expect to see happen?
The tests to pass on newer versions of libraries.

Failed on:
```
numpy 1.13.1
h5py 2.7.0
scipy 0.19.1
sklearn 0.19.0
matplotlib 2.0.2 
```

##### What did you see instead?
```
framework/PostProcessors/CrossValidations.leaveOneLabelOut: Exception in thread 0:
framework/PostProcessors/CrossValidations.leaveOneLabelOut: Traceback (most recent call last):
framework/PostProcessors/CrossValidations.leaveOneLabelOut:   File "/home/jjc/miniconda2/lib/python2.7/threading.py", line 801, in __bootstrap_inner
framework/PostProcessors/CrossValidations.leaveOneLabelOut:     self.run()
framework/PostProcessors/CrossValidations.leaveOneLabelOut:   File "/home/jjc/miniconda2/lib/python2.7/threading.py", line 754, in run
framework/PostProcessors/CrossValidations.leaveOneLabelOut:     self.__target(*self.__args, **self.__kwargs)
framework/PostProcessors/CrossValidations.leaveOneLabelOut:   File "/home/jjc/sub/raven/framework/Runners/SharedMemoryRunner.py", line 134, in <lambda>
framework/PostProcessors/CrossValidations.leaveOneLabelOut:     self.thread = threading.Thread(target = lambda q, *arg : q.append(self.functionToRun(*arg)), name = self.identifier, args=(self.subque,)+tuple(self.args))
framework/PostProcessors/CrossValidations.leaveOneLabelOut:   File "/home/jjc/sub/raven/framework/Models/PostProcessor.py", line 208, in evaluateSample
framework/PostProcessors/CrossValidations.leaveOneLabelOut:     returnValue = (Input, self.interface.run(Input))
framework/PostProcessors/CrossValidations.leaveOneLabelOut:   File "/home/jjc/sub/raven/framework/PostProcessors/CrossValidation.py", line 250, in run
framework/PostProcessors/CrossValidations.leaveOneLabelOut:     inputDict, self.CVEstimator = self.inputToInternal(inputIn, full = True)
framework/PostProcessors/CrossValidations.leaveOneLabelOut:   File "/home/jjc/sub/raven/framework/PostProcessors/CrossValidation.py", line 218, in inputToInternal
framework/PostProcessors/CrossValidations.leaveOneLabelOut:     if newInput.values().count(None) != 0:
framework/PostProcessors/CrossValidations.leaveOneLabelOut: ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

##### Do you have a suggested fix for the development team?
Same general fix as bd14004ad10147654bdc8403dd7e7c37afc2ff17

##### Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or improvement? ***improvement***
- [x] 2. Is it tagged with a priority: critical, normal or minor? ***normal***
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements? ***n/a***
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.  (Causes crash, not wrong results) ***not a defect. New version compatibility***
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.) ***yes***

Issue Reviewed by @joshua-cogliati-inl on 2017-Sept-13

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed? ***not a defect***
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.) ***not a defect***
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)? ***not a defect***
- [x] 4. If the issue is a defect, does it impact the latest stable branch? If yes, is there any issue tagged with stable (create if needed)? ***not a defect***
- [x] 5. If the issue is being closed without a merge request, has an explanation of why it is being closed been provided? ***#351***

Issue Closure Checklist Reviewed by @alfoa on 2017-Sept-13

### PR DESCRIPTION
**Pull request by joshua-cogliati-inl**
Closes #350

--------
Pull Request Description
--------
##### What issue does this change request address?
#350 

##### What are the significant changes in functionality due to this change request?
Changes one line to use "is None".

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

File: framework/PostProcessors/CrossValidation.py [hunk 1]
```diff
@@ -215,7 +215,7 @@ def inputToInternal(self, currentInp, full = False):
       #here we do not make a copy since we assume that the dictionary is for just for the model usage and any changes are not impacting outside
       newInput = currentInput
 
-    if newInput.values().count(None) != 0:
+    if True in [x is None for x in newInput.values()]:
       varName = newInput.keys()[list(newInput.values()).index(None)]
       self.raiseAnError(IOError, "The variable: ", varName, " is not exist in the input: ", currentInput.name, " which is required for model: ", cvEstimator.name)
```
Comment (wangcj05) [inline framework/PostProcessors/CrossValidation.py]: Reviewed, good to me.

File: framework/PostProcessors/CrossValidation.py [hunk 2]
```diff
@@ -215,7 +215,7 @@ def inputToInternal(self, currentInp, full = False):
       #here we do not make a copy since we assume that the dictionary is for just for the model usage and any changes are not impacting outside
       newInput = currentInput
 
-    if newInput.values().count(None) != 0:
+    if True in [x is None for x in newInput.values()]:
```
Comment (alfoa) [inline framework/PostProcessors/CrossValidation.py]: any(x is None for x in newInput.values())

Comment (joshua-cogliati-inl) [reply framework/PostProcessors/CrossValidation.py]: Changed.

**Comment by aalfonsi** [comment]
----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code. ***ok***
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts). ***n/a***
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details. ***ok***
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass. ***ok***
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True. ***n/a***
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync. ***n/a***
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done. ***#350***
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added? ***n/a***

**Comment by aalfonsi** [comment]
merging...

**Pull request merged**